UNRELEASED.md

```diff
@@ -12,4 +12,5 @@
      DO NOT LEAVE A BLANK LINE BELOW THIS PREAMBLE -->
 ### Bug fixes
 * Fixed infinite recursion in `consChain`, see #1307
-* Fix a bug where some simplified `Or` expressions were not expected by the rewriting rules, see #1285
+* Fixed a bug where some simplified `Or` expressions were not expected by the rewriting rules, see #1285
+* Fixed a bug with consistency checking in `PreproPassPartial`, see #1313
```

Comment:
Not sure if we need a release note if the bug itself wasn't released
```suggestion
```

Reply:
Wdym? It's a bug on unstable.

Reply:
To @gabrielamafra's point: release notes are meant to inform users of the released software what has changed from one release to the next. If the bug never made it into a release (i.e., a non-snapshotted version for which we released an artifact package), then there's nothing to inform users of releases about :)

Reply:
To put a point on it: we only cut releases once per week (generally) not for every commit to unstable, so just because something has made it into the trunk doesn't mean it was part of a release.

Reply:
I see your point, but if we want to go in this direction with UNRELEASED, we'd need CI to also run tests against the latest release version, so we can identify whether a bug was introduced before or after a release. The fact that I could tell which commit this bug comes from is, in general, coincidence.

Reply:
We do run tests against the latest release version! :)

You make a good case for including an item in the changelog when we don't know when the bug was introduced. But we do in fact know here.

Reply:
> Also, on an unrelated note, I have no idea how this got past our tests to begin with, see the MWE in https://github.com/informalsystems/apalache/issues/1313.

Is this because we do have an integration test that would catch this but it failed silently or because we were missing a test to cover this case?

Either way, shouldn't this PR include a fix to add such a test (or fix the silent failure), otherwise we won't be able to prevent regressions, right?

Reply:
> You make a good case for including an item in the changelog when we don't know when the bug was introduced. But we do in fact know here.
Yeah, I merged in the suggestion.

> We do run tests against the latest release version! :)

Yeah, I was a bit sloppy with the comment, what I meant was that we have very little in place to troubleshoot the source commit of a given bug (setting aside for a moment whether tracking that down is even worthwhile to begin with). TLDR: how does one, in general (not in this case), know whether to put a bugfix in the notes or not?

Reply:
We can always do git-bisect to discover when a bug was introduced. But that's not part of our process currently.

You make a good point about gray areas, but this doesn't seem to be one.